File: src/main/java/komu/demodel/domain/MoveDirection.java
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain;

public enum MoveDirection {
    UP(-1), DOWN(1);
    
    public final int delta;
   
    private MoveDirection(int delta) {
        this.delta = delta;
    }
}

File: src/main/scala/komu/demodel/domain/MoveDirection.scala
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain

sealed abstract class MoveDirection(val delta: Int)

object MoveDirection {
  case object UP extends MoveDirection(-1)
  case object DOWN extends MoveDirection(1)
}
